src/main/java/io/github/seggan/slimypowersuits/machines/ElementForge.java

```diff
@@ -0,0 +1,153 @@
+        Block dispenser = locateDispenser(b);
+        Block piston = b.getRelative(0, 2, 0);
+
+        BlockState dispenserState = PaperLib.getBlockState(dispenser, false).getState();
+        BlockData pistonData = piston.getBlockData();
+
+        if (dispenserState instanceof Dispenser && pistonData instanceof Piston
+            && ((Piston) pistonData).getFacing() == BlockFace.DOWN
+            && piston.getRelative(0, -1, 0).getType() == Material.AIR) {
+            Dispenser disp = (Dispenser) dispenserState;
+            Piston piss = (Piston) pistonData;
+            Inventory inv = disp.getInventory();
+            List<ItemStack[]> inputs = RecipeType.getRecipeInputList(this);
+
+            for (ItemStack[] input : inputs) {
+                if (isCraftable(inv, input)) {
+                    ItemStack output = RecipeType.getRecipeOutputList(this, input).clone();
+
+                    if (Slimefun.hasUnlocked(p, output, true)) {
+                        Inventory outputInv = findOutputInventory(output, dispenser, inv);
+
+                        if (outputInv != null) {
+                            craft(output, inv, outputInv);
+
+                            movePiston(piston, piss, true);
+
+                            Utils.runSync(() -> movePiston(piston, piss, false), 10L);
+
+                        } else {
+                            SlimefunPlugin.getLocalization().sendMessage(p, "machines.full-inventory", true);
+                        }
+                    }
+
+                    return;
+                }
+            }
+
+            SlimefunPlugin.getLocalization().sendMessage(p, "machines.pattern-not-found", true);
+        }
+    }
+
+    private boolean isCraftable(Inventory inv, ItemStack[] recipe) {
+        for (int j = 0; j < inv.getContents().length; j++) {
+            if (!SlimefunUtils.isItemSimilar(inv.getContents()[j], recipe[j], true)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private void craft(ItemStack output, Inventory inv, Inventory outputInv) {
+        for (int j = 0; j < 9; j++) {
+            ItemStack item = inv.getContents()[j];
+
+            if (item != null && item.getType() != Material.AIR) {
+                ItemUtils.consumeItem(item, true);
+            }
+        }
+        outputInv.addItem(output);
+    }
+
+    private Block locateDispenser(Block b) {
+        Block block = null;
+
+        if (b.getRelative(1, 0, 0).getType() == Material.DISPENSER) {
+            block = b.getRelative(1, 0, 0);
+        } else if (b.getRelative(0, 0, 1).getType() == Material.DISPENSER) {
+            block = b.getRelative(0, 0, 1);
+        } else if (b.getRelative(-1, 0, 0).getType() == Material.DISPENSER) {
+            block = b.getRelative(-1, 0, 0);
+        } else if (b.getRelative(0, 0, -1).getType() == Material.DISPENSER) {
+            block = b.getRelative(0, 0, -1);
+        }
+
+        return block;
```

Comment:
No need for this Block variable, you could just immediately return the found value in each branch.